src/dht/findpeer.js

```diff
@@ -17,10 +19,58 @@ module.exports = (send) => {
       opts = {}
     }
 
-    send.andTransform({
+    const handleResult = (res, callback) => {
+      // Inconsistent return values in the browser
+      if (Array.isArray(res)) {
+        res = res[0]
+      }
+
+      // Type 2 keys (inconsistencies between go core and js core)
+      if (res.Type !== 2 && res.type !== 2) {
+        const errMsg = `key was not found (type 2)`
+
+        return callback(errcode(new Error(errMsg), 'ERR_KEY_TYPE_2_NOT_FOUND'))
+      }
+
+      // inconsistencies between go core and js core
+      let id
+      let addrs
+
+      if (res.Responses) {
+        id = res.Responses[0].ID
+        addrs = res.Responses[0].Addrs
+      } else {
+        id = res.responses[0].id
+        addrs = res.responses[0].addrs
+      }
```

Comment:
Apologies if I wasn't clear, `interface-ipfs-core` should return lowered property names but the HTTP API should be compatible with go-ipfs.

Here I expect us to be converting "Uppered" property names to lowered form. In js-ipfs the HTTP API should take the lowered property names from core and "Upper" them to be compatible. I know this is a PITA but this is consistent with all our APIs.

Reply:
Will change it! Thanks for clarifying.